deployments/sandbox/environment.py

```diff
@@ -53,19 +53,19 @@ def env() -> Mapping[str, Optional[str]]:
         ]),
 
         'AZUL_TDR_SOURCES': ','.join([
-            'tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210219',
+            'tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210219:42',
         ]),
         **{
             f'AZUL_TDR_{catalog.upper()}_SOURCES': ','.join([
-                'tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302'
+                f'tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302:{prefix}'
             ])
-            for catalog in ('dcp2ebi', 'it2ebi')
+            for catalog, prefix in (('dcp2ebi', '42'), ('it2ebi', ''))
```

Comment:
@hannes-ucsc The IT will generate its own prefix at runtime,
https://github.com/DataBiosphere/azul/blob/28e48bb45eae0abaa4d9c4e9d72244cd4ea6e5d2/test/integration_test.py#L533
this prefix will be used to list_bundles then synth notifications. Currently when the list_bundles operation is called the listed bundles are NOT appended to any prefix that is defined in the configuration.

By adding the `source.name.prefix` from the TDRSourceName to the directly to the TDR query that is performed the prefix value that is generated from the IT is appended to the TDRSourceName prefix (`42`). This causes problems with the IT for some sources.


```
2021-04-27 09:07:59,573 INFO    MainThread: There are 3 bundle(s) with prefix '0' in source TDRSourceRef(id='c5cfdcda-995d-4f6e-94b7-e0532f6d83ca', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='hca_dev_20201023_ebiv4___20210302', is_snapshot=True)).
2021-04-27 09:07:59,574 INFO    MainThread: Selecting 64 bundles with project metadata, out of 3 candidates, using random seed 3021656021161523122.
2021-04-27 09:07:59,574 INFO    MainThread: Not enough bundles with prefix '0' in catalog 'it2ebi'. Trying a shorter prefix.
2021-04-27 09:07:59,574 INFO    MainThread: Preparing notifications for catalog 'it2ebi' and prefix ''.
2021-04-27 09:07:59,574 INFO    MainThread: Listing bundles with prefix '' in source TDRSourceRef(id='c5cfdcda-995d-4f6e-94b7-e0532f6d83ca', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='hca_dev_20201023_ebiv4___20210302', is_snapshot=True)).
2021-04-27 09:07:59,574 DEBUG   MainThread: Query: "\n            SELECT links_id, version\n            FROM broad-jade-dev-data.hca_dev_20201023_ebiv4___20210302.links\n            WHERE STARTS_WITH(links_id, '42')\n        "
2021-04-27 09:08:01,674 INFO    MainThread: There are 15 bundle(s) with prefix '' in source TDRSourceRef(id='c5cfdcda-995d-4f6e-94b7-e0532f6d83ca', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='hca_dev_20201023_ebiv4___20210302', is_snapshot=True)).
2021-04-27 09:08:01,674 INFO    MainThread: Selecting 64 bundles with project metadata, out of 15 candidates, using random seed 3021656021161523122.
2021-04-27 09:08:01,675 WARNING MainThread: Not enough bundles in catalog 'it2ebi'. The test may fail.

2021-04-27 09:08:01,676 INFO    MainThread: Preparing notifications for catalog 'it3lungmap' and prefix 'bd'.
2021-04-27 09:08:02,161 INFO    MainThread: Listing bundles with prefix 'bd' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:02,162 DEBUG   MainThread: Query: "\n            SELECT links_id, version\n            FROM broad-jade-dev-data.lungmap_dev_20210412__20210414.links\n            WHERE STARTS_WITH(links_id, '42bd')\n        "
2021-04-27 09:08:04,074 INFO    MainThread: There are 0 bundle(s) with prefix 'bd' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:04,075 INFO    MainThread: Selecting 64 bundles with project metadata, out of 0 candidates, using random seed 3021656021161523122.
2021-04-27 09:08:04,075 INFO    MainThread: Not enough bundles with prefix 'bd' in catalog 'it3lungmap'. Trying a shorter prefix.
2021-04-27 09:08:04,075 INFO    MainThread: Preparing notifications for catalog 'it3lungmap' and prefix 'b'.
2021-04-27 09:08:04,075 INFO    MainThread: Listing bundles with prefix 'b' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:04,075 DEBUG   MainThread: Query: "\n            SELECT links_id, version\n            FROM broad-jade-dev-data.lungmap_dev_20210412__20210414.links\n            WHERE STARTS_WITH(links_id, '42b')\n        "
2021-04-27 09:08:06,490 INFO    MainThread: There are 0 bundle(s) with prefix 'b' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:06,490 INFO    MainThread: Selecting 64 bundles with project metadata, out of 0 candidates, using random seed 3021656021161523122.
2021-04-27 09:08:06,490 INFO    MainThread: Not enough bundles with prefix 'b' in catalog 'it3lungmap'. Trying a shorter prefix.
2021-04-27 09:08:06,490 INFO    MainThread: Preparing notifications for catalog 'it3lungmap' and prefix ''.
2021-04-27 09:08:06,490 INFO    MainThread: Listing bundles with prefix '' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:06,490 DEBUG   MainThread: Query: "\n            SELECT links_id, version\n            FROM broad-jade-dev-data.lungmap_dev_20210412__20210414.links\n            WHERE STARTS_WITH(links_id, '42')\n        "
2021-04-27 09:08:09,485 INFO    MainThread: There are 0 bundle(s) with prefix '' in source TDRSourceRef(id='215793b0-b991-4b73-9584-1103d4072f39', name=TDRSourceName(prefix='42', project='broad-jade-dev-data', name='lungmap_dev_20210412__20210414', is_snapshot=True)).
2021-04-27 09:08:09,485 INFO    MainThread: Selecting 64 bundles with project metadata, out of 0 candidates, using random seed 3021656021161523122.
2021-04-27 09:08:09,485 WARNING MainThread: Not enough bundles in catalog 'it3lungmap'. The test may fail.
```


We might be able to figure out a prefix that fits our IT bundle requirements, but this can be tuned at a later time.

Reply:
The IT shortens the prefix until it finds enough bundles. If a configured prefix prevents the IT from finding any bundles, even with an IT prefix of `''` then the configured prefix is wrong and actually includes no bundles. So if a prefix doesn't work for a source of an IT catalog, that prefix wouldn't work for that source on a non-IT catalog either since the source wouldn't contribute any bundles to said catalog.